src/cors-proxy/init.lua

```diff
@@ -70,7 +70,7 @@ function _M:rewrite()
     ngx.req.set_method(upstream.method)
   end
 
-  ngx.req.set_uri(upstream.path)
```

Comment:
I guess we could get away with calling `ngx.unescape_uri`.

Reply:
Well, then we would need to identify whether it is escaped, and only then unescape it, otherwise, we will still be changing the original URL.
What inconvenience do you see in appending it to the upstream in `proxy_pass`?

Reply:
Well I think it is unnecessary and would like to keep most of the logic in lua - not in nginx.

And it looks like the path is always escaped.

Reply:
Unescape_uri would convert `+` into ` ` even when it is valid part of the URL.